File: archives/C#/0342. Power of Four.cs
public class Solution {
    public bool IsPowerOfFour(int num) {
        string s="1010101010101010101010101010101";
        int x=Convert.ToInt32(s,2);
        return ((num&x)!=0 && (num&(num-1))==0)?true:false;
    }
}

File: archives/20190519python/0342. Power of Four.py
class Solution(object):
    def isPowerOfFour(self, num):
        """
        :type num: int
        :rtype: bool
        """
        tmp=0b101010101010101010101010101010101010
        if num&tmp>0:
            return False
        rep=0
        while num>0:
            rep+=num&1
            num>>=1
        return rep==1
        
